src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -141,6 +141,21 @@ public async Task<bool> TryRefreshAsync()
             return true;
         }
 
+        public void SetDirty()
+        {
+            DateTimeOffset currentTime = DateTimeOffset.UtcNow;
+
+            foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers)
+            {
+                changeWatcher.CacheExpirationTime = currentTime;

```

Comment:
Can we name it CacheExpires instead of CacheExpirationTime?

Reply:
From `CacheExpires` it isn't clear what this field is meant to store. `CacheExpirationTime` makes it explicit that it stores the time when the cache expires.

---
In reply to: [438453096](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/165#discussion_r438453096) [](ancestors = 438453096)

Reply:
LastModified, Created, Expires, etc. is the common naming pattern across platforms and frameworks (see also RFC)

Reply:
We also have SetCacheExpiration, not SetCacheExpirationTime.

Reply:
`CacheExpirationTime` is a field that we populate based on current time and cache interval. It isn't set by `SetCacheExpiration` method. `Expires` as a field name seems fine, but would be more appropriate in an object that represents the cache. `CacheExpires` seems odd and I couldn't find any component that uses such a name.

Reply:
Can you find a component that uses CacheExpirationTime?

Reply:
`CacheExpires` -> `CacheExpiration`

The `SetCacheExpiration` API sets the value of `CacheExpiration` property :)

Reply:
The `SetCacheExpiration` sets the cache expiration interval (`TimeSpan`). The `CacheExpirationTime` here is different (`DateTimeOffset`).

Reply:
The current SetCacheExpiration is effectively SetCacheExpirationInterval